syntax = "proto3";
package KvStore;

message GetRequest {
    string key = 1;
}

message GetResponse {
    bool exists = 1;
    string value = 2;
}

message SetRequest {
    string key = 1;
    string value = 2;
}

message SetResponse {
    bool success = 1;
}

service KvService {
    rpc Get(KvStore.GetRequest) returns (KvStore.GetResponse);

    rpc Set(KvStore.SetRequest) returns (KvStore.SetResponse);
}